### ISSUE TITLE
Document class is unlisted in the discriminator map - using no discriminator map

### ISSUE THREAD

**Issue by josefsabl** [doctrine/mongodb-odm#2002]
### Bug Report

|    Q        |   A
|------------ | ------
| BC Break    | yes
| Version     | 2.0.0-beta2

#### Summary

Document class "Domain\XYZ\Document" is unlisted in the discriminator map although I am using no discriminator map.

Possibly related to this: https://github.com/doctrine/mongodb-odm/issues/867

#### Current behavior

Let's assume this code:

```
/**
 * @ODM\Document()
 * @ODM\InheritanceType("SINGLE_COLLECTION")
 * @ODM\DiscriminatorField("class")
 */
class DocumentA {
    /**
     * @ODM\ReferenceOne(targetDocument="DocumentA")
     * @ODM\Index
     * @var Document|null
     */
    public $parentDocument;
}

/**
 * @ODM\Document()
 */
class DocumentB extends DocumentA {
}

$docB = new DocumentB;
$docB->parentDocument = $docA;
```

Now I get the `Document class "DocumentA" is unlisted in the discriminator map` exception on a flush.

The interesting fact is that when I create the $docA in the same script as the $docB everything works.

However when I load it from the database prior setting it to the property, I get the exception.

**Comment by alcaeus** [comment]
Could you by any chance reproduce this in a test case? You can find a whole bunch of such test cases in https://github.com/doctrine/mongodb-odm/tree/master/tests/Doctrine/ODM/MongoDB/Tests/Functional/Ticket. Thanks!

**Comment by josefsabl** [comment]
Will give it a shot.

**Comment by josefsabl** [comment]
Done. I hope the clearing loaded metadata like that is okay. It simulates my problem where request that doesn't work with document A but works with document B fails.

**Comment by josefsabl** [comment]
Until the bug is fixed, I forked the ODM and just plainly disabled the notification whatsoever to enable me to work on my project.

**Comment by alcaeus** [comment]
Fix has been merged to 1.3 and master. Thanks for reporting, and more so for the failing test case!

**Comment by josefsabl** [comment]
The exception went away, but a new problem appeared. See: #2024
